lib/core/src/restore.rs

```diff
@@ -29,135 +29,156 @@ impl TxMap {
+            None => None,
         }
     }
 }
 
 pub(crate) struct RecoveredOnchainDataChainReceive {
     /// LBTC tx locking up funds by the swapper
-    lbtc_server_lockup_tx_id: Option<HistoryTxId>,
+    pub(crate) lbtc_server_lockup_tx_id: Option<HistoryTxId>,
     /// LBTC tx that claims to our wallet. The final step in a successful swap.
-    lbtc_server_claim_tx_id: Option<HistoryTxId>,
+    pub(crate) lbtc_server_claim_tx_id: Option<HistoryTxId>,
     /// BTC tx initiated by the payer (the "user" as per Boltz), sending funds to the swap funding address.
-    btc_user_lockup_tx_id: Option<HistoryTxId>,
+    pub(crate) btc_user_lockup_tx_id: Option<HistoryTxId>,
     /// BTC tx initiated by the SDK to a user-chosen address, in case the initial funds have to be refunded.
-    btc_refund_tx_id: Option<HistoryTxId>,
+    pub(crate) btc_refund_tx_id: Option<HistoryTxId>,
 }
 impl PartialSwapState for RecoveredOnchainDataChainReceive {
-    fn derive_partial_state(&self) -> PaymentState {
+    fn derive_partial_state(&self) -> Option<PaymentState> {
         match &self.btc_user_lockup_tx_id {
             Some(_) => match &self.lbtc_server_claim_tx_id {
                 Some(lbtc_server_claim_tx_id) => match lbtc_server_claim_tx_id.confirmed() {
-                    true => PaymentState::Complete,
-                    false => PaymentState::Pending,
+                    true => Some(PaymentState::Complete),
+                    false => Some(PaymentState::Pending),
                 },
                 None => match &self.btc_refund_tx_id {
                     Some(tx) => match tx.confirmed() {
-                        true => PaymentState::Failed,
-                        false => PaymentState::RefundPending,
+                        true => Some(PaymentState::Failed),
+                        false => Some(PaymentState::RefundPending),
                     },
-                    None => PaymentState::Created,
+                    None => Some(PaymentState::Pending),
                 },
             },
-            None => PaymentState::Created,
+            None => None,
         }
     }
 }
 
 pub(crate) struct RecoveredOnchainData {
-    send: HashMap<String, RecoveredOnchainDataSend>,
-    receive: HashMap<String, RecoveredOnchainDataReceive>,
-    chain_send: HashMap<String, RecoveredOnchainDataChainSend>,
-    chain_receive: HashMap<String, RecoveredOnchainDataChainReceive>,
+    pub(crate) send: HashMap<String, RecoveredOnchainDataSend>,
+    pub(crate) receive: HashMap<String, RecoveredOnchainDataReceive>,
+    pub(crate) chain_send: HashMap<String, RecoveredOnchainDataChainSend>,
+    pub(crate) chain_receive: HashMap<String, RecoveredOnchainDataChainReceive>,
 }
 
 impl LiquidSdk {
+    pub(crate) async fn get_monitored_swaps_list(&self, height: u32) -> Result<SwapsList> {
+        let send_swaps = self.persister.list_pending_send_swaps()?;
+        let receive_swaps = self.persister.list_ongoing_receive_swaps()?;
```

Comment:
Do we list here also received swaps that were expired?

Reply:
No, but the expiry will be the same as the invoice expiry, so the invoice shouldn't be valid. At this point also the MRH address can be reused from the address pool, so shouldn't be associated to the swap